pool_automation/roles/aws_manage/library/test_stateful_set.py

```diff
@@ -93,21 +103,60 @@ def test_find_ubuntu_image(ec2):
     assert 'UNSUPPORTED' not in image.description
 
 
-def test_create_instances(ec2):
+def test_AwsEC2Launcher(ec2):
+    launcher = AwsEC2Launcher()
     params = PARAMS._replace(role='test_create')
-    instances = create_instances(ec2, params, 2)
+    instances = launcher.launch(params, 2, ec2=ec2)
 
     assert len(instances) == 2
+    assert len(launcher._ids) > 0
+
     for instance in instances:
+        instance.reload()
+        assert instance.state['Name'] == 'pending'
```

Comment:
What if AWS is fast enough so state becomes `starting`? In fact I don't this assert is even needed - as far as I can see contract of launcher is:
- on `launch` return asked number of instances without any guarantees about their state
- after `wait` it's guaranteed that all instances are in some defined state

Reply:
Yes. Not actually necessary. Was a kind of debugging